src/bin/basic.rs

```diff
@@ -1,35 +1,64 @@
 //! # Basic Sample
 //!
-//! This sample demonstrates how to create a toplevel `window`, set its title, size and position, how to add a `button` to this `window` and how to connect signals with actions.
-
-#![crate_type = "bin"]
+//! This sample demonstrates how to create a toplevel `window`, set its title, size and
+//! position, how to add a `button` to this `window` and how to connect signals with
+//! actions.
 
+extern crate gio;
 extern crate gtk;
 
+use gio::prelude::*;
 use gtk::prelude::*;
 
-fn main() {
-    if gtk::init().is_err() {
-        println!("Failed to initialize GTK.");
-        return;
-    }
+use std::env::args;
+
+// make moving clones into closures more convenient
+macro_rules! clone {
```

Comment:
Seems we need add this macro to glib, or use separate crate (I seen it in this month but can't remember when).

Reply:
https://crates.io/crates/closet

Reply:
Yes, thanks.

Reply:
I was just thinking adding a new crate to `gtk-rs` with some utils like that.